bundles/org.eclipse.ltk.core.refactoring/src/org/eclipse/ltk/core/refactoring/CompositeChange.java

```diff
@@ -235,17 +233,16 @@ public void initializeValidationData(IProgressMonitor pm) {
 	@Override
 	public RefactoringStatus isValid(IProgressMonitor pm) throws CoreException {
 		RefactoringStatus result= new RefactoringStatus();
-		pm.beginTask("", fChanges.size()); //$NON-NLS-1$
+
+		SubMonitor subMonitor= SubMonitor.convert(pm, fChanges.size());
 		for (Iterator<Change> iter= fChanges.iterator(); iter.hasNext() && !result.hasFatalError();) {
 			Change change= iter.next();
-			if (change.isEnabled())
-				result.merge(change.isValid(new SubProgressMonitor(pm, 1)));
-			else
+			if (change.isEnabled()) {
+				result.merge(change.isValid(subMonitor.split(1)));
```

Comment:
is there a pattern when to use split() vs newChild()?

Reply:
newChild will not check for cancelation while split will do

Reply:
i understand the technical difference but i still don't know when i should use which. (unrelated to this PR)

Reply:
My "simple" pattern is as follows: If possible, use `split()`, otherwise use `newChild()`.
"possible" means: You are in a state in which you can safely cancel the operation (without resulting in an intermediate/unexpected system state). It makes sense to cancel as early as possible to prevent further execution when cancellation has already been requested, but of course you should ensure that you do not leave behind an invalid system state.